### PR TITLE
cockpit.spec: cockpit-{machines, ovirt} should not require libvirt metapackage

### PR DESCRIPTION
**Pull request by KKoukiou**
libvirt-daemon-kvm is sufficient.

### CODE REVIEW

**Comment by martinpitt** [comment]
libvirt-daemon-kvm is certainly sufficient, but IMHO not required. -kvm does not even exist on some architectures, so for these there at least needs to be an alternate dependency to libvirt-daemon-driver-qemu. So ideally it would be

    Requires: (libvirt-daemon-kvm or libvirt-daemon-driver-qemu or libvirt)

See http://rpm.org/user_doc/boolean_dependencies.html. Note that they don't yet work on rhel/centos 7, so this needs some `%if` guarding.

(Also, yay for our consistent package naming!)

**Comment by martinpitt** [comment]
Also, libvirt-daemon-kvm is not on any of our test images, as libvirt does not depend on it. libvirt-daemon-driver-qemu is present, though.

**Comment by KKoukiou** [comment]
Maybe we should totally skip libvirt metapackage requirement, because cockpit-machines now does not support other drivers.
```
Requires: (libvirt-daemon-kvm or libvirt-daemon-driver-qemu)
```

**Comment by martinpitt** [comment]
@KKoukiou : Sounds perfect, thanks!

**Comment by Antique** [comment]
So as I've wrote libvirt-daemon-kvm is probably the best choice, it's a meta package that will ensure that everything for KVM virtualization is there.  For comparison packages that would be installed using:

**libvirt-daemon-driver-qemu**
- libvirt-daemon
- libvirt-libs
- libvirt-daemon-driver-network
- libvirt-daemon-driver-storage-core
- /usr/bin/qemu-img
- gzip
- bzip2
- lzop
- xz

**libvirt-daemon-kvm**
- libvirt-daemon
- libvirt-daemon-driver-qemu
- libvirt-daemon-driver-interface
- libvirt-daemon-driver-network
- libvirt-daemon-driver-nodedev
- libvirt-daemon-driver-nwfilter
- libvirt-daemon-driver-secret
- libvirt-daemon-driver-storage
- qemu-kvm

As you can see, for `libvirt-daemon-driver-qemu` there is no dependency on qemu binary package and some other libvirt drivers that are usually used.  So the best option is to change the meta package `libvirt` into meta package `libvirt-daemon-kvm`.

**Comment by martinpitt** [comment]
@KKoukiou: No, you don't need to. The -qemu package should be present already, so the alternative dependency should satisfy this. Also, again: -kvm isn't a thing on e. g. aarch64 and ppc, so there must be an alternative non-KVM dependency anyway.

File: tools/cockpit.spec [hunk 1]
```diff
@@ -675,7 +675,11 @@ BuildArch: noarch
 Summary: Cockpit user interface for virtual machines
 Requires: cockpit-bridge >= %{required_base}
 Requires: cockpit-system >= %{required_base}
-Requires: libvirt
+%if 0%{?rhel}
```
Comment (martinpitt) [inline tools/cockpit.spec]: This is only required for == 7, RHEL 8 has working rich dependencies.

**Comment by Antique** [comment]
I should have write it explicitly, `libvirt-daemon-driver-qemu` alone is not sufficient because it will not install qemu-kvm package where qemu-system-x86_64 binary is so you would not be able to start any guest.

**Comment by Antique** [comment]
@martinpitt Where did you get the information that kvm ins't a thing on aarch64 or ppc64? Look at the latest koji build [https://koji.fedoraproject.org/koji/buildinfo?buildID=1141699](https://koji.fedoraproject.org/koji/buildinfo?buildID=1141699), all the architectures have libvirt-daemon-kvm package.

**Comment by KKoukiou** [comment]
So after all this discussion looks like libvirt-daemon-kvm package is available for all archs so I reverted back to the initial commit.

**Comment by martinpitt** [comment]
@Antique: So much the better, thanks. Last time I checked (but that's already two years ago), neither of these architectures had kvm support in the kernel. Also, not all of Debian's have kvm, but this PR only changes the RPM spec.

File: tools/cockpit.spec [hunk 2]
```diff
@@ -675,7 +675,11 @@ BuildArch: noarch
 Summary: Cockpit user interface for virtual machines
 Requires: cockpit-bridge >= %{required_base}
 Requires: cockpit-system >= %{required_base}
+%if 0%{?rhel} == 7
 Requires: libvirt
+%else
+Requires: libvirt-daemon-kvm or libvirt
```
Comment (martinpitt) [inline tools/cockpit.spec]: This needs to be put into parentheses, otherwise it looks like three independent dependencies to RPM.

**Comment by KKoukiou** [comment]
@martinpitt @Antique I think we have a problem here.
With this change ```libvirt-daemon-driver-storage``` and ```libvirt-daemon-driver-network``` are not brought, which means basic pools/volumes/network listing API will not work.
Either I have to install these drivers explicitly or revert this change.
What do you think?

**Pull request merged**